fix(css/ast): unit

alexander-akait opened the pull request:
Refactor `unit`, it will be `Dimension` node in future (from `<dimension-token>`). Also using `raw` give us more ability and faster codegen, because we don't need to convert them, also `kind: null` fixed

alexander-akait committed:
fix(css/ast): unit

alexander-akait committed:
fix(css/ast): unit

alexander-akait committed:
style(css/parser): fix

alexander-akait committed:
style(css/parser): fix

kdy1 committed:
fix(es): Fix bugs (#2447)

swc_ecma_parser:
 - Fix parsing of jsx escapes.

swc_ecma_minifier:
 - Don't mangle name as `const` or `let`. (#2446)

swc_ecma_transforms_react:
 - Fix handling of jsx escapes. (#2351)

kdy1 committed:
fix(es/transforms/base): Fix `resolver` (#2448)

swc_ecma_transforms_base:
 - `resolver`: Handle `AssignPatProp` from `Hoister`.

kdy1 committed:
perf(es/transforms): Reduce usage of `#[fast_path]` (#2442)

swc_ecma_ast:
 - Implement `Take` for `ObjectLit`.
 - Implement `Take` for `ArrayLit`.

swc_ecma_transforms_compat:
 - `nullish_coalescing`: Remove exponential visit.
 - `nullish_coalescing`: Migrate to `VisitMut`.
 - `optional_chaining`: Reduce exponential visit. (But it's not removed)
 - `class_properties`: Remove exponential visit.
 - `jsx_src`: Migrate to `VisitMut`.
 - `optional_chaining`: Mae it parallel.
 - `async_to_generator`: Remove `#[fast_path]`.
 - `arrow`: Remove `#[fast_path]`.
 - `destructuring`: Remove `#[fast_path]`.
 - `block_scoped_functions`:  Remove `#[fast_path]`.
 - `classes`: Remove `#[fast_path]`.

kdy1 committed:
fix(es/minifier): Fix minification of react hooks (#2450)

swc_ecma_minifier:
 - `analyze`: Track for conditional usages.
 - `collapse_vars`: Don't inline a variable if usage it conditinal.

kdy1 committed:
fix(es/codegen): Escape backtick of synthesized template literals (#2453)

swc_ecma_codegen:
 - Escape backtick of a template literal if it's synthesized.

alexander-akait committed:
fix(css/parser): Fix parsing of numbers (#2444)

kdy1 committed:
perf(es/transforms): Make transforms parallel (#2449)

swc_ecma_utils:
 - `collect_decls`: More parallel.

swc_ecma_transforms_macros:
 - Add `Parallel`, which is helper for `#[parallel]`.
 - Add `ParExplode`, which is helper for `#[parallel(explode)]`.

swc_ecma_transforms_macros:
 - Add `#[parllel]`.

swc_ecma_transforms_compat:
 - `sticky_regex`: Parallel.
 - `typeof_symbol`: Parallel.
 - `for_of`: Remove exponential visit.
 - `regenerator`: Remove exponential visit.
 - `object_spread`: Parallel.
 - `instance_of`: Parallel.
 - `duplicate_keys`: Parallel.
 - `logical_assignments`: Parallel.
 - `template_literal`: Parallel.
 - `block_scoped_functions`: Migrate to `VisitMut`.
 - `for_of`: Migrate to `VisitMut`.
 - `destructuring`: Reduce `Visit`.
 - `arrow`: Migrate to `VisitMut`.
 - `function_name`: Parallel.
 - `reserved_words`: Parallel.
 - `for_of`: Parallel.

swc_ecma_transforms_module:
 - `import_analyzer`: Migrate to `VisitMut`.

swc_ecma_transforms_react:
 - `jsx_src`: Parallel.
-  `jsx_self`: Migrate to `VisitMut`.
-  `jsx_self`: Parallel.

swc_ecma_transforms_proposal:
 - `export_default_from`: Migrate to `VisitMut`.

swc_ecma_transforms_optimization:
 - `inline_globals`: Parallel.
 - `json_parse`: Migrate to `VisitMut`.
 - `json_parse`: Parallel.

kdy1 committed:
Bump version

kdy1 committed:
test(es): Use typescript test suite as a golden testing (#2456)

kdy1 reviewed (commented):
Code changes look fine to me.
I used `Unit` to optimize performance, but actually css parsing is not time-consuming task in most cases.

kdy1 committed:
fix(common): Fix sourcemap (#2457)

swc_common:
 - Include the index to the name in `mappings` of a sourcemap.

swc:
 - Fix handling of `inputSourceMap`.

kdy1 committed:
fix(es/minifier): Check for conditional usages while inlining (#2459)

swc_ecma_minifier:
 - `hoist_props`: Check for `used_in_cond` before inlining.

kdy1 committed:
chore: Add `include` to `swc`

XiNiHa committed:
chore: Add CI steps for combined benchmark data (#2458)

kdy1 committed:
perf(es/transforms): Improve performance (#2454)

swc_ecma_transforms_macros:
 - `#[parallel]`: Run hooks for single-threaded operations.

swc_ecma_transforms_compat:
 - `shorthand`: Parallel.
 - `exponentation`: Parallel.
 - `parameters`: Parallel.
 - Recover some `#[fast_path]`

kdy1 committed:
fix(es/transforms/compat): Fix regenerator (#2460)

kdy1 committed:
fix(es/regenerator): Fix for nested try (#2463)

swc_ecma_transforms_compat:
 - `regenerator`: Use correct order for try locations array.

kdy1 committed:
chore: Publish `swc_ecma_transforms_compat`

alexander-akait committed:
refactor(css/parser): merge
